03/ukol16.py

```diff
@@ -0,0 +1,22 @@
+#Napiš program, který postupně načte od uživatele dvě čísla a jednoznakový řetězec – buď '+', '-', '*'
+#nebo '/'. Program provede na číslech příslušnou operaci.
+
+prvni = int(input('Zadej prvni cislo: '))
+operace =input('Zadej matematicke znamenko: ')
+druhe = int(input('Zadej druhe cidlo: '))
+
+if operace == '+':
+    vysledek = (f'{prvni + druhe}')
+    print(f'{prvni} + {druhe} = {vysledek}')
+elif operace == '-':
+    vysledek = (f'{prvni - druhe}')
+    print(f'{prvni} - {druhe} = {vysledek}')  
+elif operace == '*':
+    vysledek = (f'{prvni * druhe}')
+    print(f'{prvni} * {druhe} = {vysledek}')  
+elif operace == '/':
+    vysledek = (f'{prvni / druhe}')
+    print(f'{prvni} / {druhe} = {vysledek}')      
+
+
+
```

Comment:
Kratším zápisem by to šlo - ale jít na to přes `if`y je naprosto v pořádku.

Teoreticky by na to šel použít třeba docela nový konstrukt `match` (viz https://docs.python.org/3/whatsnew/3.10.html#pep-634-structural-pattern-matching), který je k dispozici od Pythonu 3.10 - pokud chceš, můžeš si o něm přečíst v dokumentaci a zkusit ho použít.

Jinak tedy `if`y jsou v pořádku, ale třeba ty těch vícero printů je trochu zbytené. Celé to ifování by šlo zkrátit na:

```python
if operace == '+':
  vysledek = prvni + druhe
elif operace == '-':
  vysledek = prvni - druhe
elif operace == '/':
  vysledek = prvni / druhe
elif operace == '*':
  vysledek = prvni * druhe

print(f'{prvni} {operace} {druhe} = {vysledek}')
```